### PR TITLE
Auto attrition selection

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
If no one sets attrition (setting it to zero still counts), then when a mission starts it will automatically set the attrition for your team.
It will default to an amount equal to the number of players on your team (limited by how much attrition you actually have).
## Why It's Good For The Game
Bald faction leader forgetting to set attrition will no longer severely fuck your team.
## Changelog
:cl:
qol: Campaign: If attrition is not manually set, it will automatically assign an amount at the start of a mission
/:cl:

### CODE REVIEW

**Comment by PibeCantina** [comment]
PRAISE

File: code/datums/gamemodes/campaign/faction_stats.dm [hunk 1]
```diff
@@ -225,15 +226,36 @@ GLOBAL_LIST_INIT(campaign_mission_pool, list(
...
+/datum/faction_stats/proc/set_attrition(amount, mob/user)
+	var/combined_attrition = total_attrition_points + active_attrition_points
+	amount = clamp(amount, 0, combined_attrition)
+	total_attrition_points = combined_attrition - amount
+	active_attrition_points = amount
+	stats_flags |= CAMPAIGN_TEAM_HAS_SET_ATTRITION
+
+	for(var/mob/living/carbon/human/faction_member AS in GLOB.alive_human_list_faction[faction])
+		faction_member.playsound_local(null, 'sound/effects/CIC_order.ogg', 30, 1)
+		to_chat(faction_member, "<span class='warning'>[user ? user : "Auto selection"] has assigned [amount] attrition points for the next mission.")
```
Comment (TiviPlus) [inline code/datums/gamemodes/campaign/faction_stats.dm:239]: This is why we use span defines come on lol

File: code/datums/gamemodes/campaign/faction_stats.dm [hunk 2]
```diff
@@ -451,14 +473,11 @@ GLOBAL_LIST_INIT(campaign_mission_pool, list(
...
-			choice = clamp(choice, 0, combined_attrition)
-			total_attrition_points = combined_attrition - choice
-			active_attrition_points = choice
-			for(var/mob/living/carbon/human/faction_member AS in GLOB.alive_human_list_faction[faction])
-				faction_member.playsound_local(null, 'sound/effects/CIC_order.ogg', 30, 1)
-				to_chat(faction_member, "<span class='warning'>[user] has assigned [choice] attrition points for the next mission.")
-			update_static_data_for_all_viewers()
+			//check again so you can't just hold the window open
+			if((current_mode.current_mission?.mission_state != MISSION_STATE_NEW) && (current_mode.current_mission?.mission_state != MISSION_STATE_LOADED))
+				to_chat(user, "<span class='warning'>Current mission already ongoing, unable to assign more personnel at this time.")
```
Comment (TiviPlus) [inline code/datums/gamemodes/campaign/faction_stats.dm:478]: AGAIN DUDE FIX YOUR SPANS

File: code/datums/gamemodes/campaign/faction_stats.dm [hunk 3]
```diff
@@ -474,7 +493,7 @@ GLOBAL_LIST_INIT(campaign_mission_pool, list(
 			if(current_mode.current_mission?.mission_state != MISSION_STATE_FINISHED)
 				to_chat(user, "<span class='warning'>Current mission still ongoing!")
 				return
-			if(!(stats_flags & MISSION_SELECTION_ALLOWED))
+			if(!(stats_flags & CAMPAIGN_TEAM_MISSION_SELECT_ALLOWED))
 				to_chat(user, "<span class='warning'>The opposing side has the initiative, win a mission to regain it.")
```
Comment (TiviPlus) [inline code/datums/gamemodes/campaign/faction_stats.dm:497]: LUMIIIIIIIIIIIIIIII

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/gamemodes/campaign/faction_stats.dm (hunk 3 above)
Comment (Lumipharon) [reply code/datums/gamemodes/campaign/faction_stats.dm:497]: gonna do it every day until you like it.

**Pull request merged**